pkg/fqdn/dnspoller.go

```diff
@@ -60,10 +60,14 @@ type DNSPoller struct {
 	// config is a copy from when this instance was initialized.
 	// It is read-only once set
 	config Config
+
+	// DNSHistory is the collection of still-valid DNS responses intercepted
+	// for the poller
+	DNSHistory *DNSCache
 }
 
 // NewDNSPoller creates an initialized DNSPoller. It does not start the controller (use .Start)
-func NewDNSPoller(config Config, ruleManager *RuleGen) *DNSPoller {
+func NewDNSPoller(config Config, ruleManager *RuleGen, DNSHistory *DNSCache) *DNSPoller {
```

Comment:
Addressed in the comment in `daemon/fqdn.go`

Reply:
I don't think I understand (but I did reply to https://github.com/cilium/cilium/pull/7629/files#r274884582). The poller needs cache specific to it, with a perHostIP limit. This can't be the global cache (which this PR also removes). We can create the global cache in `BootstrapFQDN()` since it now needs to know the minTTL but the poller doesn't need a reference to that.